### ISSUE TITLE
Sparse operations in `SparseInversion` occur after updating the ekp object.

### ISSUE THREAD

**Issue by ilopezgp** [CliMA/EnsembleKalmanProcesses.jl#113]
The SparseInversion update performs the following operations

```
# Sparse EKI
    cov_vv = vcat(hcat(cov_uu, cov_ug), hcat(cov_ug', cov_gg))
    H_u = hcat(1.0 * I(size(u)[1]), fill(FT(0), size(u)[1], size(g)[1]))
    H_g = hcat(fill(FT(0), size(g)[1], size(u)[1]), 1.0 * I(size(g)[1]))

    H_uc = H_u[ekp.process.uc_idx, :]

    cov_vv_inv = cov_vv \ (1.0 * I(size(cov_vv)[1]))

    # Loop over ensemble members to impose sparsity
    Threads.@threads for j in 1:(ekp.N_ens)
        # Solve a quadratic programming problem
        u[:, j] = sparse_qp(ekp, v[j, :], cov_vv_inv, H_u, H_g, y[:, j], H_uc = H_uc)

        # Threshold the results if needed
        if ekp.process.threshold_eki
            u[ekp.process.uc_idx, j] =
                u[ekp.process.uc_idx, j] .* (abs.(u[ekp.process.uc_idx, j]) .> ekp.process.threshold_value)
        end

        # Add small noise to constrained elements of u
        u[ekp.process.uc_idx, j] += rand(
            ekp.rng,
            MvNormal(zeros(size(ekp.process.uc_idx)[1]), ekp.process.reg * I(size(ekp.process.uc_idx)[1])),
        )
    end
```

after the ekp object has been updated,

```
push!(ekp.u, DataContainer(u, data_are_columns = true))
push!(ekp.g, DataContainer(g, data_are_columns = true))
```

This means that these manipulations do not really affect the advanced parameters. The initial block of operations copied in this issue should be done before pushing new DataContainers to the ekp object.

**Comment by jinlong83** [comment]
It had no bug as it was consistent with previous version of DataContainer in terms of memory. I'll change the implementation accordingly to make it consistent with the updated version of DataContainer in #114

**Comment by ilopezgp** [comment]
Fixed by #119
